lib/class-wp-rest-pattern-directory-controller.php

```diff
@@ -0,0 +1,281 @@
+	 */
+	public function get_items( $request ) {
+		switch ( $request->get_route() ) {
+			case '/__experimental/pattern-directory/search':
+				$query_args = array(
+					'action' => 'query_patterns',
+					'search' => sanitize_text_field( $request['term'] ),
+				);
+
+				break;
+
+			case '/__experimental/pattern-directory':
+			default:
+				$category   = (int) $request->get_param( 'category' );
+				$query_args = array(
+					'action' => 'get_patterns',
+				);
+
+				if ( $category ) {
+					$query_args['pattern-categories'] = $category;
+				}
+
+				break;
+		}
+
+		$api_url = add_query_arg(
+			$query_args,
+			'http://api.wordpress.org/patterns/1.0/'
+		);
+
+		if ( wp_http_supports( array( 'ssl' ) ) ) {
+			$api_url = set_url_scheme( $api_url, 'https' );
+		}
+
+		$response     = wp_remote_get( $api_url );
+		$raw_patterns = json_decode( wp_remote_retrieve_body( $response ) );
+
+		if ( is_wp_error( $response ) ) {
+			$response->add_data( array( 'status' => 500 ) );
+
+			return $response;
+		}
+
+		$prepared_patterns = array();
+
+		if ( $raw_patterns ) {
+			foreach ( $raw_patterns as $pattern ) {
+				$prepared_patterns[] = $this->prepare_item_for_response( $pattern, $query_args['action'] );
+			}
+		}
+
+		return rest_ensure_response( $prepared_patterns );
+	}
+
+	/**
+	 * Parse pattern metadata for a pattern, and prepare it for an API response.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param array  $raw_pattern A pattern from api.wordpress.org, before any changes.
+	 * @param string $action      The type of request that resulted in the raw patterns. Either 'get_patterns' or 'query_patterns'.
+	 *
+	 * @return array
+	 */
+	public function prepare_item_for_response( $raw_pattern, $action ) {
+		switch ( $action ) {
+			default:
+			case 'get_patterns':
+				$prepared_pattern                 = (array) $raw_pattern;
+				$prepared_pattern['category_ids'] = wp_list_pluck( $raw_pattern->{'_embedded'}->{'wp:term'}[0], 'id' );
+				break;
+
+			case 'query_patterns':
+				$prepared_pattern                 = (array) $raw_pattern->{'_embedded'}->self[0];
+				$prepared_pattern['category_ids'] = $prepared_pattern['pattern-categories'];
+				break;
+		}
+
+		$prepared_pattern['title']          = $prepared_pattern['title']->rendered;
+		$prepared_pattern['content']        = $prepared_pattern['content']->rendered;
+		$prepared_pattern['description']    = $prepared_pattern['meta']->wpop_description;
+		$prepared_pattern['viewport_width'] = $prepared_pattern['meta']->wpop_viewport_width;
+
+		// Remove irrelevant properties.
+		$prepared_pattern = array_intersect_key( $prepared_pattern, $this->get_item_schema()['properties'] );
+
+		return $prepared_pattern;
+	}
+
+	/**
+	 * Retrieves the pattern's schema, conforming to JSON Schema.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		$this->schema = array(
```

Comment:
The purpose of the `$schema` property is meant to cache the schema, so we should make sure to early return if it is specified. We need to call `add_additional_fields_schema` after the cache.

Reply:
👍🏻 , fixed in 611dc4d